### PR TITLE
WIP refactor which command

### PR DESCRIPTION
**Pull request by umar-ahmed**
## Description

- Add chalk dependency
- Add error and success coloured log functions
- Extract logic for getting active yarn version into helper function

### DevQA Prep

- Please `make install`, this PR uses a new package

### DevQA Steps

- Run yvm which in the various use cases
	- Verify that errors are shown in red
	- Verify that message for matching version in .yvmrc is shown in green

### Comments:

- This is a WIP
- Replaces https://github.com/tophat/yvm/pull/122

### CODE REVIEW

File: src/commands/which.js [hunk 1]
```diff
@@ -1,56 +1,26 @@
...
-            )
-
-            const pathVersion = matchedVersion.toString().replace(/v/g, '')
+const whichCommand = () => {
+    getActiveVersion()
+        .catch(() => log('Sorry, yarn in NOT installed.'))
+        .then(activeVersion => {
+            log(`Found yarn version: ${activeVersion}`)
             const rcVersion = getRcFileVersion()
-            log(`Currently on yarn version ${pathVersion}`)
```
Comment (jakebolam) [inline src/commands/which.js]: All looks great. Should this call `getSplitVersionAndArgs` (albeit poorly named) to determine the 'should be used version' i.e. take into account the default version and RC file version

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
   "author": "Top Hat <opensource@tophat.com>",
   "license": "Apache-2.0",
   "dependencies": {
+    "chalk": "^2.4.1",
```
Comment (jakebolam) [inline package.json]: :+1

File: src/commands/which.js [hunk 3]
```diff
@@ -1,56 +1,26 @@
...
-        if (element.startsWith(versionRootPath(yvmPathToUse))) {
-            const versionRegex = /(v\d+\.?\d*\.?\d*)/gm
-            const matchedVersion = element.match(versionRegex)
-            log.info(
-                `matched yvm version: ${matchedVersion} in PATH ${element}`,
-            )
-
-            const pathVersion = matchedVersion.toString().replace(/v/g, '')
+const whichCommand = () => {
+    getActiveVersion()
```
Comment (jakebolam) [inline src/commands/which.js]: [nb] asynchronous/await

File: src/util/version.js [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
+const shell = require('shelljs')
+
 const log = require('./log')
 const { yvmPath: defaultYvmPath } = require('./utils')
 
 function isValidVersionString(version) {
     return /^\d+\.\d+\.\d+$/.test(version)
 }
 
+function getActiveVersion() {
```
Comment (jakebolam) [inline src/util/version.js]: [NB] get yarn actual shell version. (There's so many versions in this app)

File: src/yvm.js [hunk 5]
```diff
@@ -15,101 +15,110 @@ const invalidCommandLog = () =>
     )
 
 /* eslint-disable global-require,prettier/prettier */
-argParser
-    .description('Yarn Version Manager');
+argParser.description('Yarn Version Manager')
```
Comment (jakebolam) [inline src/yvm.js]: Lint!! :)

**Comment by iamogbz** [comment]
@umar-ahmed when you get back to this, could your conflict resolution include https://github.com/tophat/yvm/pull/275#discussion_r245470771

**Pull request closed without merging**